lib/rules/no-undef.js

```diff
@@ -110,6 +113,7 @@ module.exports = function(context) {
             variable = new escope.Variable(node.local.name, scope);
         variable.defs.push(node);
```

Comment:
It seems like that could work. But haven't we tried in the past to avoid letting rules modify the AST?